### PR TITLE
Add server testcase for engine/user.go

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Add server testcase for engine/user.go

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: calendar/engine/user_test.go [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+
+func TestUser_String(t *testing.T) {
+	tests := []struct {
+		name       string
+		user       *User
+		assertions func(t *testing.T, actualString string)
+	}{
+		{
+			name: "User with the Mattermost user object",
+			user: &User{
```
Comment (raghavaggarwal2308) [inline calendar/engine/user_test.go:215]: We can use the existing util for getting this

Comment (raghavaggarwal2308) [reply calendar/engine/user_test.go:215]: We just have to modify that a little

File: calendar/engine/user_test.go [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+			},
+		},
+		{
+			name: "failed to delete event subscription",
+			setupMock: func() {
+				mscalendar.client = mockClient
+				mscalendar.actingUser = &User{MattermostUserID: MockRemoteUserID}
+				mockWelcomer.EXPECT().AfterDisconnect(MockMMUserID).Return(nil)
+				mockStore.EXPECT().LoadUser(MockMMUserID).Return(&store.User{Settings: store.Settings{EventSubscriptionID: MockEventSubscriptionID}}, nil).Times(1)
+				mockStore.EXPECT().LoadSubscription(MockEventSubscriptionID).Return(nil, nil)
```
Comment (raghavaggarwal2308) [inline calendar/engine/user_test.go:380]: Add .Times() everywhere

File: calendar/engine/user_test.go [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+				mscalendar.AdminUserIDs = "mockAdminID1,mockAdminID2"
+			},
+			assertions: func(result bool, err error) {
+				require.NoError(t, err)
+				require.Equal(t, true, result)
+			},
+		},
+		{
+			name:             "error checking system admin",
+			mattermostUserID: "mockMMUserID",
```
Comment (raghavaggarwal2308) [inline calendar/engine/user_test.go:515]: Can use one of the existing constants

File: calendar/engine/user_test.go [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+			},
+			assertions: func(result *store.Settings, err error) {
+				require.Error(t, err)
+				require.ErrorContains(t, err, "error filtering user")
+			},
+		},
+		{
+			name: "Successfully get user settings",
+			setupMock: func() {
+				mockUser.User = &store.User{Settings: store.Settings{GetConfirmation: false}, Remote: &remote.User{ID: "testRemoteID"}}
```
Comment (raghavaggarwal2308) [inline calendar/engine/user_test.go:583]: Can use one of the existing constant for ID

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**